lib/mongoose_push/pools.ex

```diff
@@ -38,7 +38,7 @@ defmodule MongoosePush.Pools do
   @doc "Return random worker name for given service and with given `:mode` set"
   @spec select_worker(MongoosePush.service, MongoosePush.mode) :: atom
   def select_worker(service, mode) do
-    [pool | _] = pools_by_mode(service, mode)
+    [pool | _] = __MODULE__.pools_by_mode(service, mode)
```

Comment:
The point here is to call the function explicitly from `__MODULE__` module, because when you load mock, the original module name changes, so `__MODULE__` points to mock module, while if you call function "locally", you would call the function without even touching the mock, which is not desirable.